.github/workflows/ci.yml

```
@@ -0,0 +1,96 @@
+name: CI checks
+
+on: 
+  pull_request:
+    types: [synchronize, opened, reopened, ready_for_review]
+
+## `actions-rs/toolchain@v1` overwrite set to false so that 
+## `rust-toolchain` is always used and the only source of truth.
+
+jobs:
+  test:
+    if: github.event.pull_request.draft == false
+
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # We don't need to test across multiple platforms yet
+        # os: [ubuntu-latest, windows-latest, macOS-latest]
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          override: false
+      - name: Run tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --verbose --release --all --all-features
+
+  build:
+    if: github.event.pull_request.draft == false
+
+    name: Build target ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+          - wasm32-unknown-unknown
+          - wasm32-wasi
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          override: false
+      - name: Add target
+        run: rustup target add ${{ matrix.target }}
+      - name: cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --all-features
+
```

Añadiría compilación para benches. Creo que es algo así, pero comprueba antes de mergear. 
```suggestion
      - name: Build benchmarks
        continue-on-error: false
        with:
          command: bench
          args: --no-run
          
```

---

LI'll add this with the Benchmarks PR as now we don't have them.

While addressing #12